service/history/historyEngine.go

```diff
@@ -78,6 +78,10 @@ const (
 	replicationTimeout                        = 30 * time.Second
 	contextLockTimeout                        = 500 * time.Millisecond
 
+	// pollMutableStateBufferDuration should be smaller than minPollMutableStateTimeout
```

Comment:
nit: this can be enforced if `minPollMutableStateTimeout` is declared like `minPollMutableStateTimeout = pollMutableStateBufferDuration * 2`